Formula/opentracing-cpp.rb

```diff
@@ -0,0 +1,146 @@
+        }
+        // Extract and Inject a span context.
+        {
+          std::unordered_map<std::string, std::string> text_map;
+          TextMapCarrier carrier(text_map);
+          auto err = tracer->Inject(parent_span->context(), carrier);
+          assert(err);
+          auto span_context_maybe = tracer->Extract(carrier);
+          assert(span_context_maybe);
+          auto span = tracer->StartSpan("propagationSpan",
+                                        {ChildOf(span_context_maybe->get())});
+        }
+        // You get an error when trying to extract a corrupt span.
+        {
+          std::unordered_map<std::string, std::string> text_map = {
+              {"x-ot-span-context", "123"}};
+          TextMapCarrier carrier(text_map);
+          auto err = tracer->Extract(carrier);
+          assert(!err);
+          assert(err.error() == span_context_corrupted_error);
+          // How to get a readable message from the error.
+          std::cout << "Example error message: \"" << err.error().message() << "\"\n";
+        }
+        parent_span->Finish();
+        tracer->Close();
+        std::cout << "\nRecorded spans as JSON:\n\n";
+        std::cout << oss.str() << "\n";
+        return 0;
+      }
+    EOS
+    # This is from example/tutorial/text_map_carrier.h, minus empty lines
+    (testpath/"text_map_carrier.h").write <<~'EOS'
+      #ifndef LIGHTSTEP_TEXT_MAP_CARRIER
+      #define LIGHTSTEP_TEXT_MAP_CARRIER
+      #include <opentracing/propagation.h>
+      #include <string>
+      #include <unordered_map>
+      using opentracing::TextMapReader;
+      using opentracing::TextMapWriter;
+      using opentracing::expected;
+      using opentracing::string_view;
+      class TextMapCarrier : public TextMapReader, public TextMapWriter {
+       public:
+        TextMapCarrier(std::unordered_map<std::string, std::string>& text_map)
+            : text_map_(text_map) {}
+        expected<void> Set(string_view key, string_view value) const override {
+          text_map_[key] = value;
+          return {};
+        }
+        expected<void> ForeachKey(
+            std::function<expected<void>(string_view key, string_view value)> f)
+            const override {
+          for (const auto& key_value : text_map_) {
+            auto result = f(key_value.first, key_value.second);
+            if (!result) return result;
+          }
+          return {};
+        }
+       private:
+        std::unordered_map<std::string, std::string>& text_map_;
+      };
+      #endif  // LIGHTSTEP_TEXT_MAP_CARRIER
+    EOS
+    (testpath/"CMakeLists.txt").write <<~'EOS'
+      cmake_minimum_required(VERSION 3.1)
+      project(tracingtest)
+      set(CMAKE_CXX_STANDARD 11)
+      find_package(OpenTracing REQUIRED)
+      add_executable(tutorial-example tutorial-example.cpp)
+      target_link_libraries(tutorial-example opentracing)
+      target_link_libraries(tutorial-example opentracing_mocktracer)
+    EOS
+    system "cmake", "."
```

Comment:
Let's not require `cmake` in the text. Use: `system ENV.cxx` like other formulas do.